### PR TITLE
pull request

### PR DESCRIPTION
**Pull request by Agula2018**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -117,6 +117,18 @@
...
+			<groupId>mysql</groupId>
+			<artifactId>mysql-connector-java</artifactId>
+			<scope>runtime</scope>
+		</dependency>
+		<dependency>
+			<groupId>javax.servlet</groupId>
+			<artifactId>jsp-api</artifactId>
+			<version>2.0</version>
+			<scope>provided</scope>
+		</dependency>
```
Comment (neonelect) [inline pom.xml]: You don't need that

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.logging.Logger;
+
+@RestController
+public class AirlyController {
+
+    private static final Logger logger = Logger.getLogger(AirlyController.class.getName());
+    private final String MAIN_URL = "https://airapi.airly.eu/";
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java]: Remove that for now

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import com.agnieszka.piotrowska.weatherCheckApp.service.ParsingService;
+import com.google.gson.JsonParseException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.logging.Logger;
+
+@RestController
+public class AirlyController {
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java:10]: Please add @RequestMapping and a domain path

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+     * @param lat
+     * @param lng
+     * @param maxDistanceKM
+     * @param maxResults
+     * @param model
+     * @return
+     * @throws JsonParseException
+     */
+
+    @RequestMapping("v2/installations/")
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java]: remove that

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    @GetMapping("nearest/")
+    public Installations handleNearestInstallation(@RequestParam("lat") double lat,
+                                                   @RequestParam("lng") double lng,
+                                                   @RequestParam("maxDistanceKM") double maxDistanceKM,
+                                                   @RequestParam("maxResults") int maxResults,
+                                                   Model model) throws JsonParseException {
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java]: You don't need model

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        Installations installations = (Installations) parsingService.parse(MAIN_URL);
+        logger.info("installation request by: " + lat + lng + maxDistanceKM + maxResults);
+        model.addAttribute("installation", installations);
+        return installations;
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java]: Please return a mocked value like "OK"

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    }
+
+    @GetMapping("/v2/installations/{installationId}")
+    public Installations handleInstallationId(@PathVariable("installationId") int installationId,
+                                              Model model) throws JsonParseException {
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java]: You don't need model

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+        logger.info("installation request by id: " + installationId);
+        Installations installations = (Installations) parsingService.parse(MAIN_URL);
+        model.addAttribute("installationId", installations);
+        return installations;
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java]: Please return mocked value

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+     * @param installationId
+     * @param model
+     * @return
+     * @throws JsonParseException
+     */
+
+    @GetMapping("/v2/measurements/installation")
+    public MeasurementHistory detailedMeasurement(@RequestParam("indexType") String indexType,
+                                                  @RequestParam("installationId") int installationId,
+                                                  Model model) throws JsonParseException {
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java]: You don't need model

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+        logger.info("detailed measurements for an installation " + indexType + installationId);
+        MeasurementHistory measurementHistory = (MeasurementHistory) parsingService.parse(MAIN_URL);
+        model.addAttribute("measurementHistory" + measurementHistory);
+        return measurementHistory;
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java]: Please return mocked value

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+    }
+
+    @GetMapping("/v2/measurements/nearest")
+    public MeasurementHistory detailedMeasurementNearest(@RequestParam("indexType") String indexType,
+                                                         @RequestParam("lat") double lat,
+                                                         @RequestParam("lng") double lng,
+                                                         @RequestParam("maxDistanceKM") double maxDistanceKM,
+                                                         Model model) throws JsonParseException {
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java]: You don't need model

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+        logger.info("detailed measurements for an installation " + indexType + lat + lng + maxDistanceKM);
+        MeasurementHistory measurementHistory = (MeasurementHistory) parsingService.parse(MAIN_URL);
+        model.addAttribute("measurementHistory" + measurementHistory);
+        return measurementHistory;
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java]: Please return mocked value

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+    }
+
+    @GetMapping("/v2/measurements/point")
+    public MeasurementHistory detailedMeasurementPoint(@RequestParam("indexType") String indexType,
+                                                       @RequestParam("lat") double lat,
+                                                       @RequestParam("lng") double lng,
+                                                       Model model) throws JsonParseException {
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java]: You don't need model

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+        logger.info("detailed measurements for an installation " + indexType + lat + lng);
+        MeasurementHistory measurementHistory = (MeasurementHistory) parsingService.parse(MAIN_URL);
+        model.addAttribute("measurementHistory" + measurementHistory);
+        return measurementHistory;
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java]: Please return mocked value

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/model/Address.java [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@NoArgsConstructor
+@AllArgsConstructor
+@JsonIgnoreProperties
+@Getter
+@Setter
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/model/Address.java]: Consider checking if it's really necessary to have all these annotations

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/model/Arguments.java [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@JsonIgnoreProperties
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/model/Arguments.java]: Consider checking if it's really necessary to have all these annotations

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/model/Arguments.java [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+public class Arguments {
+
+    public String apiKey;
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/model/Arguments.java]: This won't be needed

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/model/Index.java [hunk 18]
```diff
@@ -0,0 +1,20 @@
+package com.agnieszka.piotrowska.weatherCheckApp.model;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/model/Index.java]: Consider checking if it's really necessary to have all these annotations

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/model/Installations.java [hunk 19]
```diff
@@ -0,0 +1,23 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@JsonIgnoreProperties
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/model/Installations.java]: Consider checking if it's really necessary to have all these annotations

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/model/Installations.java [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
+public class Installations {
+
+    private Integer id;
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/model/Installations.java]: please use long if ID is a number

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/model/LocatedMeasurement.java [hunk 21]
```diff
@@ -1,11 +1,19 @@
...
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
 
-@Data
+@JsonIgnoreProperties
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
+@Setter
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/model/LocatedMeasurement.java]: Consider checking if it's really necessary to have all these annotations

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/model/MeasurementHistory.java [hunk 22]
```diff
@@ -0,0 +1,20 @@
...
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import java.util.List;
+
+@JsonIgnoreProperties
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/model/MeasurementHistory.java]: Consider checking if it's really necessary to have all these annotations

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/model/SingleMeasurement.java [hunk 23]
```diff
@@ -1,39 +1,23 @@
...
+import lombok.NoArgsConstructor;
+import lombok.Setter;
 import java.util.List;
-@Data
+
+@JsonIgnoreProperties
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
+@Setter
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/model/SingleMeasurement.java]: Consider checking if it's really necessary to have all these annotations

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/model/Sponsor.java [hunk 24]
```diff
@@ -0,0 +1,21 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@JsonIgnoreProperties
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/model/Sponsor.java]: Consider checking if it's really necessary to have all these annotations

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/model/Sponsor.java [hunk 25]
```diff
@@ -0,0 +1,21 @@
...
+public class Sponsor {
+
+    private Integer sponsorId;
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/model/Sponsor.java]: Please use long if ID is a number

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/model/Standards.java [hunk 26]
```diff
@@ -0,0 +1,21 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@JsonIgnoreProperties
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
+@Setter
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/model/Standards.java]: Consider checking if it's really necessary to have all these annotations

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/model/Value.java [hunk 27]
```diff
@@ -0,0 +1,17 @@
...
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@JsonIgnoreProperties
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/model/Value.java]: Consider checking if it's really necessary to have all these annotations

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/service/ParsingService.java [hunk 28]
```diff
@@ -0,0 +1,12 @@
+package com.agnieszka.piotrowska.weatherCheckApp.service;
+
+public interface ParsingService {
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/service/ParsingService.java]: Does it makes sense to have a Interface here?

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/web/BeanController.java [hunk 29]
```diff
@@ -0,0 +1,14 @@
+package com.agnieszka.piotrowska.weatherCheckApp.web;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.client.RestTemplate;
+
+@Configuration
+public class BeanController {
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/web/BeanController.java]: It's configuration not a controller :)

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/web/MainController.java [hunk 30]
```diff
@@ -0,0 +1,17 @@
+package com.agnieszka.piotrowska.weatherCheckApp.web;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.client.RestTemplate;
+
+@Controller
+public class MainController {
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/web/MainController.java]: Please remove this class for now

**Review by neonelect — CHANGES_REQUESTED** [review]
Check the comments

File: pom.xml (hunk 1 above)
Comment (Agula2018) [reply pom.xml]: Removed, done :).

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java (hunk 2 above)
Comment (Agula2018) [reply src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java]: Ok, done.

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java (hunk 3 above)
Comment (Agula2018) [reply src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java:10]: ok, done :)

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java (hunk 4 above)
Comment (Agula2018) [reply src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java]: Ok, done.

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java (hunk 5 above)
Comment (Agula2018) [reply src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java]: ok :)

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java (hunk 6 above)
Comment (Agula2018) [reply src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java]: ok :)

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java (hunk 7 above)
Comment (Agula2018) [reply src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java]: ok :)

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java (hunk 8 above)
Comment (Agula2018) [reply src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java]: ok :)

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java (hunk 9 above)
Comment (Agula2018) [reply src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java]: ok :)

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java (hunk 10 above)
Comment (Agula2018) [reply src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java]: ok :)

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java [hunk 31]
```diff
@@ -1,101 +1,54 @@
...
 import com.agnieszka.piotrowska.weatherCheckApp.service.ParsingService;
 import com.google.gson.JsonParseException;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.*;
 
-import java.util.logging.Logger;
-
 @RestController
+@RequestMapping("airapi.airly.eu/v2/")
```
Comment (neonelect) [inline src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java]: please leave only v2/ prefix

**Review by neonelect — CHANGES_REQUESTED** [review]
Please edit .gitignore file to omit the .idea/ and target/ directories

File: src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java (hunk 31 above)
Comment (Agula2018) [reply src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java]: What if I have 2 different prefixes? eg. 
@RequestMapping("v2/installations") and 
@RequestMapping("v2/measurement")?
Should I create a new Controller class for eg. @RequestMapping("v2/measurement")?
Thank you for reply :)

Comment (neonelect) [reply src/main/java/com/agnieszka/piotrowska/weatherCheckApp/controller/AirlyController.java]: Yeah you should then have 2 different domains so two classes: InstallationsController and MeasurementController

**Review by neonelect — APPROVED** [review]

**Pull request merged**